### PR TITLE
Add go get mason_client to Docker image

### PR DESCRIPTION
**Pull request by utka**
There will be second PR to remove it from istio/prow/mason_lib.sh

### CODE REVIEW

**Comment by sebastienvas** [comment]
the dockerfile is the server. The client needs to use the mason client. That won't work.

**Comment by utka** [comment]
Ok, is this one correct?

**Comment by sebastienvas** [comment]
Is prowbazel still used ? (Apparently yes for)
The one for istio is istio-builder I think, so you can update https://github.com/istio/test-infra/blob/master/docker/istio/shared/tools/install-golang.sh#L22
 and https://github.com/istio/test-infra/blob/master/scripts/tools/linux-install-golang#L21 for prowbazel but you ll need to rebuild all the images and update all configs.

**Comment by sebastienvas** [comment]
also what is the reason for doing this ? The reason of calling go get was to always get the latest version as this version should match the deployment. Adding it to the image should be fine, just note that when you'll update the mason client, you also have to update the job configs to roll it out. It is probably safer that way, but also more work for you.

**Comment by utka** [comment]
This is the problem which happens sometimes during tests: https://github.com/istio/test-infra/issues/1199

File: docker/prowbazel/Dockerfile [hunk 1]
```diff
@@ -45,6 +45,7 @@ RUN sudo chmod +rx /usr/local/bin/entrypoint
 RUN mkdir -p /home/bootstrap/go/src
 ENV GOPATH=/home/bootstrap/go
 ENV PATH ${GOPATH}/bin:${PATH}
+RUN go get istio.io/test-infra/boskos/cmd/mason_client
```
Comment (fejta) [inline docker/prowbazel/Dockerfile:48]: Each RUN and ENV command creates a new docker layer. Ideally we should try to combine as many of these as possible so we can minimize layers.

**Review by fejta — APPROVED** [review]
/lgtm
/hold

`/hold cancel` if you're ready for this to merge

**Comment by geeknoid** [comment]
@sebastienvas @fejta @cjwagner 

Given that Maria's left the project, what should we do with this PR? Do you guys want to take this over the finish line?

**Comment by geeknoid** [comment]
Closing. Please reopen if needed.

**Pull request closed without merging**